packer/progressbar.go

```diff
@@ -15,26 +15,28 @@ func ProgressBarConfig(bar *pb.ProgressBar, prefix string) {
 	bar.Prefix(prefix)
 }
 
-var defaultUiProgressBar = &uiProgressBar{}
+var defaultUiProgressBar = &UiProgressBar{}
 
-// uiProgressBar is a self managed progress bar singleton.
-// decorate your struct with a *uiProgressBar to
+// UiProgressBar is a self managed progress bar singleton.
+// decorate your struct with a *UiProgressBar to
 // give it TrackProgress capabilities.
-// In TrackProgress if uiProgressBar is nil
+// In TrackProgress if UiProgressBar is nil
 // defaultUiProgressBar will be used as
 // the progress bar.
-type uiProgressBar struct {
+type UiProgressBar struct {
+	Noop bool
```

Comment:
If we remove the singleton pattern then a nil UiProgressBar could be a noop progress bar:

```go
var noOpProgressBar * UiProgressBar = nil
```

Reply:
I'm not following how this would be used. Can you push up a commit making the change you're thinking of here, or pair with me on it later this week?

Reply:
I pushed a tiny commit to do that ! We can pair on it when you login today ! 🙂